experimental/juvix/contract.jvx

```diff
@@ -0,0 +1,198 @@
+Map = Data.Map
+
+-- common module interfaces for data structures in Common?
+MonoidMeasure : Common.Measure
+MonoidMeasure =
+  type T = Nat
+
+  -- (<>) is normal semi-group  type class function
+  (<>)    = (<>)
+  measure = id
+
+
+-- need OrdMap, as modules that one goes to can't be modules?
+Accounts = Map.OrdMap Data.String MonoidMeasure
+
+-- we send in the string module, as it has a total order
+-- function named compare, that is also part of a general type class
+Token =
+  type Address  = s : String.T {String.length s == 36}
+
+  type Storage = {
+    total-supply : Nat.T;
+    accounts     : Accounts.T { Accounts.measure-value == total-supply }
+  }
+
+  empty-storage : Storage
+  empty-storage = {
+    total-supply = 0;
```

Comment:
Why `;` instead of `,` on records? I think we might want to reserve `;` for possible later use as monadic sequence.

Reply:
I think `;` is fine here. This doesn't ruin the use of `;` for monadic sequence, as these are very different contexts!

Though we should probably discuss this more in depth, as per pros and cons

Reply:
Why not just `,` like Haskell?

I think `;` is confusing since it is very different than monadic sequence.

Reply:
well in OCaml and F*, they use `;` for records. It kind of reminds me of C structs.

Though, since we are using , for tuples and for lists, I will change it to `,`